syne_tune/optimizer/schedulers/searchers/searcher.py

```diff
@@ -595,15 +649,38 @@ def _generate_remaining_candidates(self) -> List[dict]:
         return remaining_candidates
 
     def _generate_all_candidates_on_grid(self) -> List[dict]:
-        # Get hp values that are specified as Domain
+        '''
+        This function generates all configurations on grid.
```

Comment:
Maybe change the wording to make more clear what this function does. For example: '_Generate all configurations to be evaluated by placing a regular, equally spaced grid over the configuration space._'